src/fsverity/ioctl.rs

```diff
@@ -71,8 +75,33 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
     }
 
     if digest.digest_algorithm != digest_algorithm || digest.digest_size != digest_size {
-        Err(std::io::Error::from(std::io::ErrorKind::InvalidData))?
+        Err(Errno::ILSEQ)
     } else {
         Ok(digest.digest)
     }
 }
+
+/// Like [`fs_ioc_measure_verity`] but does not fail if the file does not have fsverity enabled.
+pub fn fs_ioc_measure_verity_optional<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<Option<H>> {
+    match fs_ioc_measure_verity(fd) {
+        Ok(e) => Ok(Some(e)),
+        // See e.g. `lcfs_fd_measure_fsverity()`
+        Err(Errno::NODATA | Errno::NOSYS | Errno::NOTTY) => Ok(None),
+        Err(e) => Err(e),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::fsverity::Sha256HashValue;
+
+    #[test]
+    fn test_measure_verity_opt() -> anyhow::Result<()> {
+        let tf = tempfile::tempfile()?;
+        assert!(fs_ioc_measure_verity_optional::<_, Sha256HashValue>(&tf)
+            .unwrap()
```

Comment:
`?` here, since you're already in a `Return`-returning function.

Reply:
There's a bit of an idiom here - in tests specifically I use `unwrap()` on the APIs we actually want to test because it gives us precise line numbers.

It isn't relevant here, but it is in other places where one might be invoking the API under test multiple times, and if you use `?` then on a test case failure you'd be wondering which specific one failed.

I didn't create this idiom, it came from some Rust code Luca Bruno maintained IIRC. Do other people do it? Probably...

Reply:
I like this logic.  Thanks for explaining.

Reply:
See also, though, (in case you don't know it) the `backtrace` feature of `anyhow`.

Reply:
Yeah, but by default you don't get backtraces out of `cargo test` either way. Of course you can, this is just all about improving the default case for what happens when a test fails.

Reply:
Ya.  I definitely like your approach better.